Applications/Telemetry/telemetry.py

```diff
@@ -29,59 +29,99 @@
+    telemetry to a specific unit with this application. This function runs
+    in an infinite loop continually querying for data.
+
+    :param config: telemetry.ini ConfigParser object
+    :return: Nothing
     """
     logger.info('Starting telemetry_worker thread')
 
     # initialize variables
     stations = {}
+    count = 0
 
     # Initialize proxy object
     proxy = faradaybasicproxyio.proxyio()
 
     # Initialize Faraday parser
     faradayParser = telemetryparser.TelemetryParse()  # Add logger?
 
-    # Open configuration file
-    dbFilename = config.get("DATABASE", "FILENAME")
+    try:
+        # Open configuration file
+        dbFilename = config.get("DATABASE", "FILENAME")
 
-    # Pragmatically create descriptors for each Faraday connected to Proxy
-    count = config.getint("TELEMETRY", "UNITS")
+        # Pragmatically create descriptors for each Faraday connected to Proxy
+        count = config.getint("TELEMETRY", "UNITS")
+
+    except ConfigParser.Error as e:
+            #  Error reading in configs so get stuck in infinite loop indicating problem
+            while True:
+                logger.error("ConfigParse.Error: " + str(e))
+                time.sleep(1)
 
     for num in range(count):
-        callsign = config.get("TELEMETRY", "UNIT" + str(num) + "CALL").upper()
-        nodeid = config.get("TELEMETRY", "UNIT" + str(num) + "ID")
+        try:
+            callsign = config.get("TELEMETRY", "UNIT" + str(num) + "CALL").upper()
+            nodeid = config.get("TELEMETRY", "UNIT" + str(num) + "ID")
+
+        except ConfigParser.Error as e:
+            #  Error reading in configs so get stuck in infinite loop indicating problem
+            while True:
+                logger.error("ConfigParse.Error: " + str(e))
+                time.sleep(1)
+
         stations["UNIT" + str(num) + "CALL"] = callsign
         stations["UNIT" + str(num) + "ID"] = nodeid
         telemetryDicts[str(callsign) + str(nodeid)] = deque([], maxlen=1000)
 
-    # check for data on telemetry port, if True place into deque
-    while(1):
+    #  Check for data on telemetry port with infinite loop.
+    while True:
          for radio in range(count):
             callsign = stations["UNIT" + str(num) + "CALL"]
             nodeid = stations["UNIT" + str(num) + "ID"]
             data = proxy.GET(str(callsign), str(nodeid), int(proxy.TELEMETRY_PORT))
 
-            # Iterate through each packet and unpack into dictionary
-            if data != None:
+            if type(data) is dict:
+                #  A dict means something is wrong with GET, print error JSON
+                logger.info(data["error"])
+
+            elif data == None:
+                #  No data is available from Proxy
+                logger.debug("telemetryworker data GET response = None")
+
+            else:
+                # Iterate through each packet and unpack into dictionary
+                logger.debug("Proxy data: " + repr(data))
```

Comment:
If valid data: Get list of HTTP 200 JSON dictionaries of data
If error from proxy: Get single HTTP 404 JSON dictionary item
Should never be None except if never set.